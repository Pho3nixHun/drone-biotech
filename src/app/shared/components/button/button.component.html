@let vM = vm();

<button
    class="btn flex flex-row justify-center"
    [disabled]="disabled()"
    [ngClass]="{
        'btn-primary': !vM.secondary,
        'btn-secondary': vM.secondary,
        'btn-outline': vM.variant === 'outline',
        'btn-text': vM.variant === 'ghost',
        'btn-active': active(),
        'btn-disabled': disabled(),
    }"
>
    <ng-content select="[span, mat-icon]" />
</button>
