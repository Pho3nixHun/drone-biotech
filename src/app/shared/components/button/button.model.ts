import { WithMatIcon } from '@interfaces/mat-icon.enum';
import { WithRouterLink } from '@interfaces/with-router-link.interface';
import { WithTextNode } from '@interfaces/with-text-node.interface';

export type State = 'active' | 'disabled' | 'default';
type Variant = 'fill' | 'ghost' | 'outline';

export interface BaseButtonVM {
    variant: Variant;
    state?: State;
    secondary?: boolean;
}

interface ButtonVMWithTextNode extends BaseButtonVM, WithTextNode {
    type: 'withText';
}
interface ButtonVMWithMatIcon extends BaseButtonVM, WithMatIcon {
    type: 'withIcon';
}

export type ButtonVM = ButtonVMWithTextNode | ButtonVMWithMatIcon;

export type ButtonVMWithRouterLink = ButtonVM & WithRouterLink;
