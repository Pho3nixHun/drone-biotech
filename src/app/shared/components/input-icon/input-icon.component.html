<span class="label-text my-auto flex items-center">
    <mat-icon>{{ icon() }}</mat-icon>
</span>
