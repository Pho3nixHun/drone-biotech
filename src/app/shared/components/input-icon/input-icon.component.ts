import { Component, input } from '@angular/core';
import { MatIconModule } from '@angular/material/icon';
import { MatIcon } from '@interfaces/mat-icon.enum';

@Component({
    selector: 'app-input-icon',
    imports: [MatIconModule],
    templateUrl: './input-icon.component.html',
})
export class InputIconComponent {
    public icon = input.required<MatIcon>();
}
