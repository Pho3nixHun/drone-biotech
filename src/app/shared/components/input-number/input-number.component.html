<div *transloco="let t" class="my-2">
    @let vM = vm();
    <span class="flex flex-row justify-between items-end label-text">
        <label [for]="vM.id">{{ t(vM.labelKey) }}</label>
        <ng-content select="app-button" />
    </span>
    <div
        class="space-x-3 flex items-center input"
        [ngClass]="{ 'is-invalid': invalid() }"
    >
        <ng-content select="[leading]" />
        <input
            class="grow"
            type="number"
            [id]="vM.id"
            [value]="value()"
            [disabled]="disabled()"
            [placeholder]="t(vM.placeholderKey)"
            (blur)="markAsTouched()"
            (input)="changed($event)"
            [readonly]="vM.readonly"
            autocomplete="off"
        />
        <ng-content select="[trailing]" />
    </div>
    <div class="grid grid-cols-2 items-baseline helper-text">
        <span>
            <ng-content select="[leftAssistiveText]" />
        </span>
        <span class="text-right">
            <ng-content select="[rightAssistiveText]" />
        </span>
    </div>
</div>
