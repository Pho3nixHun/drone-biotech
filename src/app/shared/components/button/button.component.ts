import { Component, input } from '@angular/core';
import { ButtonVM } from './button.model';
import { NgClass } from '@angular/common';

@Component({
    selector: 'app-button',
    imports: [NgClass],
    templateUrl: './button.component.html',
})
export class ButtonComponent {
    public readonly vm = input.required<ButtonVM>();
    public readonly disabled = input<boolean>();
    public readonly active = input<boolean>();
}
