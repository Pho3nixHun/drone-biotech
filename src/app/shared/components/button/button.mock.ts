import { MatIcon } from '@interfaces/mat-icon.enum';
import { ButtonVM } from './button.model';

export const enMock = {
    text: 'txt',
};

export const mockPrimaryButton: ButtonVM = {
    type: 'withText',
    textKey: enMock.text,
    variant: 'fill',
};

export const mockSecondaryButton: ButtonVM = {
    type: 'withText',
    textKey: enMock.text,
    variant: 'outline',
    secondary: true,
};

export const mockFillButton: ButtonVM = {
    type: 'withText',
    textKey: enMock.text,
    variant: 'ghost',
};
export const mockGhostButton: ButtonVM = {
    type: 'withText',
    textKey: enMock.text,
    variant: 'ghost',
};
export const mockOutlineButton: ButtonVM = {
    type: 'withText',
    textKey: enMock.text,
    variant: 'outline',
};

export const mockTextButton: ButtonVM = {
    type: 'withText',
    textKey: enMock.text,
    variant: 'outline',
};
export const mockIconButton: ButtonVM = {
    type: 'withIcon',
    icon: MatIcon.ADD,
    variant: 'outline',
};
