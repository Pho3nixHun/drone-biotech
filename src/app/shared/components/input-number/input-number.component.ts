import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
import { Component, forwardRef, input, signal } from '@angular/core';
import { TranslocoModule } from '@jsverse/transloco';
import { ButtonXVM } from '@components/button/button.model';
import { MatIcon } from '@interfaces/mat-icon.enum';
import { NgClass } from '@angular/common';

@Component({
    selector: 'app-input-number',
    imports: [TranslocoModule, NgClass],
    providers: [
        {
            provide: NG_VALUE_ACCESSOR,
            useExisting: forwardRef(() => InputNumberComponent),
            multi: true,
        },
    ],
    templateUrl: './input-number.component.html',
})
export class InputNumberComponent implements ControlValueAccessor {
    public readonly vm = input.required<InputNumberVM>();
    public readonly invalid = input<boolean>(false);
    protected readonly disabled = signal<boolean>(false);
    protected readonly onTouched = signal<(() => void) | null>(null);
    protected readonly value = signal<number | null>(null);
    protected readonly onChange = signal<
        ((value: number | null) => void) | null
    >(null);

    protected markAsTouched() {
        const onTouched = this.onTouched();
        if (onTouched) onTouched();
    }

    protected changed(event: Event) {
        const target = event.target;
        if (!(target instanceof HTMLInputElement)) return;

        this.value.set(target.valueAsNumber);
        const onChange = this.onChange();
        if (onChange) onChange(target.valueAsNumber);
    }

    writeValue(value: number | null): void {
        this.value.set(value);
    }
    registerOnChange(fn: (value: number | null) => void): void {
        this.onChange.set(fn);
    }
    registerOnTouched(fn: () => void): void {
        this.onTouched.set(fn);
    }
    setDisabledState(isDisabled: boolean): void {
        this.disabled.set(isDisabled);
    }
}

export interface InputNumberVM {
    placeholderKey: string;
    labelKey: string;
    id: string;
    readonly: boolean;
}

export interface InputNumberXVM extends InputNumberVM {
    buttonXVM?: ButtonXVM;
    leadingIcon?: MatIcon;
    trailingIcon?: MatIcon;
}
