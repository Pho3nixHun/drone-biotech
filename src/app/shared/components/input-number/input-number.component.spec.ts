import { ComponentFixture, TestBed } from '@angular/core/testing';
import { InputNumberComponent, InputNumberXVM } from './input-number.component';
import { MatIcon } from '@interfaces/mat-icon.enum';
import { Component, DebugElement, inject, input, signal } from '@angular/core';
import { ReactiveFormsModule, FormBuilder } from '@angular/forms';
import { ButtonComponent } from '@components/button/button.component';
import { InputIconComponent } from '@components/input-icon/input-icon.component';
import { InputTextXVM } from '@components/input-text/input-text.component';
import { getTranslocoModule } from 'transloco-testing.module';
import { By } from '@angular/platform-browser';

@Component({
    imports: [
        InputIconComponent,
        InputNumberComponent,
        ButtonComponent,
        ReactiveFormsModule,
    ],
    template: `
        @let vM = vm();
        <app-input-number [vm]="vM" [invalid]="!valid()" [formControl]="fc">
            @if (vM.buttonXVM) {
                <app-button [vm]="vM.buttonXVM" />
            }
            @if (vM.leadingIcon) {
                <app-input-icon leading [icon]="vM.leadingIcon" />
            }
            @if (vM.trailingIcon) {
                <app-input-icon trailing [icon]="vM.trailingIcon" />
            }
            <span leftAssistiveText>{{ vM.requiredAssistiveTextKey }}</span>
            <span rightAssistiveText>{{ vM.requiredAssistiveTextKey }}</span>
            <div>should not be displayed</div>
        </app-input-number>
    `,
})
class TestHostComponent {
    private readonly fb = inject(FormBuilder);
    public vm = input.required<InputTextXVM>();
    public readonly valid = signal<boolean>(true);
    public readonly fc = this.fb.control('');
}

describe('InputNumberComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;
    let debugEl: DebugElement;
    let component: InputNumberComponent;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [
                TestHostComponent,
                getTranslocoModule({
                    langs: { en: enMock },
                    translocoConfig: {
                        availableLangs: ['en'],
                        defaultLang: 'en',
                    },
                }),
            ],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
        debugEl = fixture.debugElement.query(By.css('app-input-number'));
        component = debugEl.componentInstance;
        fixture.componentInstance.valid.set(true);
    });

    // Unit testing
    it('should write value via writeValue', () => {
        // Arrange
        component.writeValue(7);

        // There is no need to act

        // Assert
        expect(component['value']()).toBe(7);
    });

    // Unit testing
    it('should call onTouched when markAsTouched is called', () => {
        // Arrange
        const spy = jest.fn();
        component.registerOnTouched(spy);

        // Act
        component['markAsTouched']();

        // Assert
        expect(spy).toHaveBeenCalled();
    });

    // Snapshot testing
    it('should render the template correctly', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVM);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });

    // Snapshot testing
    it('should render the template correctly when it has valid state', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVM);
        fixture.componentInstance.valid.set(true);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });

    // Snapshot testing
    it('should render the template correctly when it has invalid state', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVM);
        fixture.componentInstance.valid.set(false);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });

    // Snapshot testing
    it('should render the template correctly if the button is provided in the vm', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVMWithButton);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });

    // Snapshot testing
    it('should render the template correctly if the leading icon is provided in the vm', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVMWithLeadingIcon);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });

    // Snapshot testing
    it('should render the template correctly if the trailing icon is provided in the vm', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVMWithTrailingIcon);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });

    // Snapshot testing
    it('should render the template correctly if everything is provided in the vm', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVMFull);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });

    // Snapshot testing
    it('should render the template if the control is enabled', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVM);
        fixture.componentInstance.fc.enable();

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });

    // Snapshot testing
    it('should render the template if the control is disabled', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVM);
        fixture.componentInstance.fc.disable();

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
    // Snapshot testing
    it('should render the template correctly if it is not readonly', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVMNotReadonly);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
    // Snapshot testing
    it('should render the template correctly if it is readonly', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVMReadonly);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});

const enMock = {
    label: 'lab',
    placeholder: 'place',
};

const mockVM: InputNumberXVM = {
    id: 'id',
    labelKey: enMock.label,
    placeholderKey: enMock.placeholder,
    readonly: false,
};

const mockVMWithButton: InputNumberXVM = {
    id: 'id',
    labelKey: enMock.label,
    placeholderKey: enMock.placeholder,
    readonly: false,
    buttonXVM: {
        icon: MatIcon.ADD,
        variant: 'fill',
    },
};

const mockVMWithLeadingIcon: InputNumberXVM = {
    id: 'id',
    labelKey: enMock.label,
    placeholderKey: enMock.placeholder,
    readonly: false,
    leadingIcon: MatIcon.ADD,
};

const mockVMWithTrailingIcon: InputNumberXVM = {
    id: 'id',
    labelKey: enMock.label,
    placeholderKey: enMock.placeholder,
    readonly: false,
    trailingIcon: MatIcon.ADD,
};
const mockVMFull: InputNumberXVM = {
    id: 'id',
    labelKey: enMock.label,
    placeholderKey: enMock.placeholder,
    readonly: false,
    trailingIcon: MatIcon.ADD,
    leadingIcon: MatIcon.CLOSE,
    buttonXVM: {
        icon: MatIcon.ADD,
        variant: 'fill',
    },
};
const mockVMNotReadonly: InputNumberXVM = {
    id: 'id',
    labelKey: enMock.label,
    placeholderKey: enMock.placeholder,
    readonly: false,
};
const mockVMReadonly: InputNumberXVM = {
    id: 'id',
    labelKey: enMock.label,
    placeholderKey: enMock.placeholder,
    readonly: true,
};
