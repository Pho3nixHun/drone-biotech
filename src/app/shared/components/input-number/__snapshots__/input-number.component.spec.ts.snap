// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`InputNumberComponent should render the template correctly 1`] = `
<div
  id="root2"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template correctly if everything is provided in the vm 1`] = `
<div
  id="root8"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <app-button>
          <button
            class="btn flex flex-row justify-center btn-primary"
          />
        </app-button>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <app-input-icon
          leading=""
        >
          <span
            class="label-text my-auto flex items-center"
          >
            <mat-icon
              aria-hidden="true"
              class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
              data-mat-icon-type="font"
              role="img"
            >
              close
            </mat-icon>
          </span>
        </app-input-icon>
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <app-input-icon
          trailing=""
        >
          <span
            class="label-text my-auto flex items-center"
          >
            <mat-icon
              aria-hidden="true"
              class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
              data-mat-icon-type="font"
              role="img"
            >
              add
            </mat-icon>
          </span>
        </app-input-icon>
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template correctly if it is not readonly 1`] = `
<div
  id="root11"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template correctly if it is readonly 1`] = `
<div
  id="root12"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          readonly=""
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template correctly if the button is provided in the vm 1`] = `
<div
  id="root5"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <app-button>
          <button
            class="btn flex flex-row justify-center btn-primary"
          />
        </app-button>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template correctly if the leading icon is provided in the vm 1`] = `
<div
  id="root6"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <app-input-icon
          leading=""
        >
          <span
            class="label-text my-auto flex items-center"
          >
            <mat-icon
              aria-hidden="true"
              class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
              data-mat-icon-type="font"
              role="img"
            >
              add
            </mat-icon>
          </span>
        </app-input-icon>
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template correctly if the trailing icon is provided in the vm 1`] = `
<div
  id="root7"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <app-input-icon
          trailing=""
        >
          <span
            class="label-text my-auto flex items-center"
          >
            <mat-icon
              aria-hidden="true"
              class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
              data-mat-icon-type="font"
              role="img"
            >
              add
            </mat-icon>
          </span>
        </app-input-icon>
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template correctly when it has invalid state 1`] = `
<div
  id="root4"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input is-invalid"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template correctly when it has valid state 1`] = `
<div
  id="root3"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template if the control is disabled 1`] = `
<div
  id="root10"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          disabled=""
          id="id"
          placeholder="place"
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;

exports[`InputNumberComponent should render the template if the control is enabled 1`] = `
<div
  id="root9"
  ng-version="20.1.7"
>
  <app-input-number
    class="ng-untouched ng-pristine ng-valid"
  >
    <div
      class="my-2"
    >
      <span
        class="flex flex-row justify-between items-end label-text"
      >
        <label
          for="id"
        >
          lab
        </label>
        <!--container-->
      </span>
      <div
        class="space-x-3 flex items-center input"
      >
        <!--container-->
        <input
          autocomplete="off"
          class="grow"
          id="id"
          placeholder="place"
          type="number"
        />
        <!--container-->
      </div>
      <div
        class="grid grid-cols-2 items-baseline helper-text"
      >
        <span>
          <span
            leftassistivetext=""
          >
            
          </span>
        </span>
        <span
          class="text-right"
        >
          <span
            rightassistivetext=""
          >
            
          </span>
        </span>
      </div>
    </div>
    <!--container-->
  </app-input-number>
</div>
`;
